### PR TITLE
roachpb,serverpb: protobuf fixups

### PR DESCRIPTION
**Pull request by knz**
Cosmetic

### CODE REVIEW

**Comment by knz** [comment]
@irfansharif confirms that the behavior is OK and thus no backport is needed

**Comment by irfansharif** [comment]
https://github.com/cockroachdb/cockroach/issues/55435#issuecomment-707045507

File: pkg/server/serverpb/admin.proto [hunk 1]
```diff
@@ -454,7 +454,8 @@ message DecommissionStatusRequest {
 message DecommissionRequest {
   repeated int32 node_ids = 1 [(gogoproto.customname) = "NodeIDs",
                                (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/roachpb.NodeID"];
-  cockroach.kv.kvserver.storagepb.MembershipStatus target_membership = 2;
+  reserved 2;
+  cockroach.kv.kvserver.storagepb.MembershipStatus target_membership = 3;
```
Comment (tbg) [inline pkg/server/serverpb/admin.proto:458]: This even creates an incompatibility, no? Shouldn't we remove this commit?

Comment (knz) [reply pkg/server/serverpb/admin.proto:458]: irfan convinced me to remove it, yes

**Comment by knz** [comment]
removed the commit, reworked the PR desc. RFAL

**Review by irfansharif — APPROVED** [review]
LGTM. The diffs to ErrorDetail look like they apply even before 20.2, but I'm not sure if we care about them enough. Could be worth adding a comment there to make sure we don't forget to update the reserved list as the error value list sees changes.

**Comment by knz** [comment]
Checking reserved lists is part of the review checklist for any protobuf change.

**Comment by knz** [comment]
bors r=irfansharif,tbg

**Pull request merged**